### ISSUE TITLE
Name "number" blocked even after whitelisting

### ISSUE THREAD

**Issue by refugeindhamma** [aj3423/SpamBlocker#40]
Hi, really nice to have a new app to stop spam calls & SMS other than "Yet Another Call Blocker".

I am receiving OTPs from accounts that send SMS but they are being blocked by SpamBlocker even after I whitelisted it.

I have set ```.*``` to block all numbers, then ```.*SAMSUNG.*``` in both ```Number Rules``` and ```Message Rules``` to whitelist it. However, it is still being blocked. I have also tried ```SAMSUNG``` and ```\bSAMSUNG\b``` and it also does not work.

Is this a bug or am I using regex wrongly?

**Comment by aj3423** [comment]
Do you set the  `.*SAMSUNG.*` with a higher priority? The whitelisted rules should have higher priority to override the `.*`

**Comment by refugeindhamma** [comment]
I set them as the same priority. It worked for the phone calls having the same priority as ```.*``` so I thought it would work for SMS too. Anyway, thanks for the solution.

**Comment by aj3423** [comment]
>  It worked for the phone calls having the same priority as .*

That depends on which rule was created first in the database, it's unreliable and may work randomly, it should be based on priority.

**Comment by refugeindhamma** [comment]
The rules for the phone calls were created first, then ```.*``` was last. After that, I noticed the SMS were not coming through so I created the SMS rules, which came after the ```.*``` rule. 

Thanks for the explanation. I have changed the priorities to make them work reliably.
